app/src/main/java/ch/epfl/skysync/components/ChatText.kt

```diff
@@ -54,10 +54,17 @@ import ch.epfl.skysync.ui.theme.veryLightGreen
  * @param paddingValues Padding values to apply around the chat interface.
  */
 @Composable
-fun ChatText(messages: List<ChatMessage>, onSend: (String) -> Unit, paddingValues: PaddingValues) {
+fun ChatText(
+    messages: List<ChatMessage>,
+    onSend: (String) -> Unit,
+    paddingValues: PaddingValues,
+    connectivityStatus: ConnectivityStatus
+) {
   Column(modifier = Modifier.padding(paddingValues).imePadding()) {
     ChatTextBody(messages, modifier = Modifier.weight(1f))
-    ChatInput(onSend)
+    if (connectivityStatus.isOnline()) {
```

Comment:
Instead of displaying the button if online you could disable the button when offline

Reply:
Good to know.
Thanks ;)